GlaaVM native error on micronaut-vertx-pg-client

### Expected Behavior

I would expect that dependency `micronaut-vertx-pg-client` add all the required GraalVM configuration in order to be able to compile my application in native mode. 

### Actual Behaviour

Given a simple app with the following dependencies:

```
dependencies {
    implementation('com.ongres.scram:client:2.1')
    implementation("io.micronaut.sql:micronaut-vertx-pg-client")
    implementation("io.micronaut.rxjava2:micronaut-rxjava2")
    implementation("io.micronaut:micronaut-runtime")
    runtimeOnly("ch.qos.logback:logback-classic")
    implementation("io.micronaut:micronaut-inject")
    implementation ('io.micronaut:micronaut-management')
}
```

App:

```
package com.example;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import io.reactivex.Single;

import io.vertx.reactivex.pgclient.PgPool;
import io.vertx.reactivex.sqlclient.Row;
import io.vertx.reactivex.sqlclient.RowIterator;
import jakarta.inject.Inject;

@Controller("/postgres")
public class HelloWorldController {

    @Inject
    PgPool pgClient;

    @Get("/hello")
    @Produces(MediaType.TEXT_PLAIN)
    public String index() {
        return "Hello World";
    }

    @Get
    @Produces(MediaType.TEXT_PLAIN)
    public Single<String> getPostgresVersion() {
        return  pgClient.query("SELECT version()")
                .rxExecute()
                .map(rows -> {
                    RowIterator<Row> rowIt = rows.iterator();
                    if (rowIt.hasNext()) {
                        return rowIt.next().getString("version");
                    }

                    return "";
                });
    }
}

```

application.yml

```
micronaut:
  application:
    name: micronaut-demo
  server:
    port: 8080
vertx:
  pg:
    client:
      port: 5432
      host: localhost
      database: mydb
      user: user
      password: topsecret
      maxSize: 5
```

When I compile my app with GraalVM `./gradlew clean nativeCompile ` 

native-image.properties

```
Args = --initialize-at-run-time=io.vertx.pgclient.impl.codec.DataTypeCodec \
       --trace-class-initialization=io.vertx.core.logging.LoggerFactory \
       --allow-incomplete-classpath \
       --enable-all-security-services \
       -H:IncludeResources=logback.xml|application.yml \
       -H:Name=rxjava-app \
       -H:Class=com.example.Application \
       -H:ReflectionConfigurationResources=${.}/reflection.json \
       -H:+ReportUnsupportedElementsAtRuntime
```

reflection.json

```
[
  {
    "name": "io.vertx.pgclient.impl.codec.DataTypeCodec",
    "allDeclaredConstructors": true,
    "allDeclaredMethods": true
  }
]
```

and then make a query to `localhost:8080/postgres` I am getting the following error:

```
.18:33:38.396 [vert.x-eventloop-thread-13] ERROR i.m.http.server.RouteExecutor - Unexpected error occurred: Could not initialize class io.vertx.pgclient.impl.codec.DataTypeCodec
java.lang.NoClassDefFoundError: Could not initialize class io.vertx.pgclient.impl.codec.DataTypeCodec

```

If I swap the graalVM config property `initialize-at-run-time` to `initialize-at-build-time` then I am getting the following compile error:

```
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of io.vertx.core.logging.SLF4JLogDelegate are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=io.vertx.core.logging.SLF4JLogDelegate.

``` 

Because looks like pgClient internally is using a Log library that needs to be initialized at runtime but it also seems that DataTypeCodec needs to be  initialized at build time, so looks like a chicken-egg problem




### Steps To Reproduce

Just create a hello world app like this one:

```
package com.example;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import io.reactivex.Single;

import io.vertx.reactivex.pgclient.PgPool;
import io.vertx.reactivex.sqlclient.Row;
import io.vertx.reactivex.sqlclient.RowIterator;
import jakarta.inject.Inject;

@Controller("/postgres")
public class HelloWorldController {

    @Inject
    PgPool pgClient;

    @Get("/hello")
    @Produces(MediaType.TEXT_PLAIN)
    public String index() {
        return "Hello World";
    }

    @Get
    @Produces(MediaType.TEXT_PLAIN)
    public Single<String> getPostgresVersion() {
        return  pgClient.query("SELECT version()")
                .rxExecute()
                .map(rows -> {
                    RowIterator<Row> rowIt = rows.iterator();
                    if (rowIt.hasNext()) {
                        return rowIt.next().getString("version");
                    }

                    return "";
                });
    }
}

```

and try to compile with GraalVM and make a Query to `/postgres`



### Environment Information

openjdk 17.0.5 2022-10-18
OpenJDK Runtime Environment GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08)
Fedora 37



### Example Application

_No response_

### Version

3.8.7

--- a/settings.gradle
+++ b/settings.gradle
@@ -65,6 +65,7 @@ include 'tests:hibernate:hibernate-reactive-postgres'
 
 include 'tests:jooq-tests:jooq-jdbc-postgres'
 include 'tests:jooq-tests:jooq-r2dbc-postgres'
+include 'tests:vertx-pg-client-native'
 
 include 'tests:jdbc-ucp-tests:jdbc-ucp-oracle'
 include 'tests:jdbc-dbcp-tests:jdbc-dbcp-postgres'

--- a/tests/vertx-pg-client-native/build.gradle
+++ b/tests/vertx-pg-client-native/build.gradle
@@ -1,0 +1,42 @@
+plugins {
+    id 'io.micronaut.build.internal.testing-base'
+    id 'io.micronaut.application'
+}
+
+micronaut {
+    importMicronautPlatform = false
+    runtime "netty"
+    testRuntime "junit"
+}
+
+dependencies {
+    runtimeOnly mnLogging.logback.classic
+    implementation mn.micronaut.http.client
+    implementation projects.micronautVertxPgClient
+
+    testImplementation mnTest.micronaut.test.core
+    testImplementation mnTest.micronaut.test.junit5
+
+    micronautBoms(platform(mn.micronaut.core.bom))
+    micronautBoms(platform(mnData.micronaut.data.bom))
+}
+
+application {
+    mainClass = "example.Application"
+}
+
+configurations.all {
+    resolutionStrategy.preferProjectModules()
+}
+
+graalvmNative {
+    toolchainDetection = false
+    binaries {
+        all {
+            buildArgs.add("--report-unsupported-elements-at-runtime")
+        }
+    }
+    metadataRepository {
+        enabled = true
+    }
+}

--- a/tests/vertx-pg-client-native/src/main/java/example/Application.java
+++ b/tests/vertx-pg-client-native/src/main/java/example/Application.java
@@ -1,0 +1,10 @@
+package example;
+
+import io.micronaut.runtime.Micronaut;
+
+public class Application {
+
+    public static void main(String[] args) {
+        Micronaut.run(Application.class);
+    }
+}

--- a/tests/vertx-pg-client-native/src/main/java/example/PgCodecController.java
+++ b/tests/vertx-pg-client-native/src/main/java/example/PgCodecController.java
@@ -1,0 +1,15 @@
+package example;
+
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+import io.vertx.pgclient.impl.codec.DataTypeCodec;
+
+@Controller("/postgres")
+final class PgCodecController {
+
+    @Get("/codec")
+    String codec() {
+        return DataTypeCodec.LDT_PLUS_INFINITY.getYear() > 0
+            && DataTypeCodec.LDT_MINUS_INFINITY.getYear() < 0 ? "ok" : "invalid";
+    }
+}

--- a/tests/vertx-pg-client-native/src/test/java/example/PgCodecNativeTest.java
+++ b/tests/vertx-pg-client-native/src/test/java/example/PgCodecNativeTest.java
@@ -1,0 +1,22 @@
+package example;
+
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@MicronautTest
+class PgCodecNativeTest {
+
+    @Inject
+    @Client("/")
+    HttpClient client;
+
+    @Test
+    void canInitializeVertxPgCodec() {
+        assertEquals("ok", client.toBlocking().retrieve("/postgres/codec"));
+    }
+}

--- a/tests/vertx-pg-client-native/src/test/resources/logback.xml
+++ b/tests/vertx-pg-client-native/src/test/resources/logback.xml
@@ -1,0 +1,10 @@
+<configuration>
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+    </appender>
+    <root level="info">
+        <appender-ref ref="STDOUT" />
+    </root>
+</configuration>

